src/persistence/mongoose/User/queries.test.js

```diff
@@ -0,0 +1,41 @@
+import { factory } from 'factory-girl';
+
+import UserFactory from '../../../../test/factories/user'; // eslint-disable-line no-unused-vars
+import queries from './queries'
+
+import { mongooseConnect, config } from '../../../index';
+
+const db = mongooseConnect(config);
+let userQueries = queries(db); // eslint-disable-line no-unused-vars
+
+describe('Users queries', () => {
+
+  describe('create', () => {
+    it('can create a user', async () => {
+      const payload = await factory.attrs('User');
+
+      const user = await userQueries.create({ payload });
+
+      expect(user).toBeDefined()
+
+      expect(user).toHaveProperty('firstname')
+      expect(user).toHaveProperty('lastname')
+      expect(user).toHaveProperty('uuid')
+      expect(user['firstname']).toEqual(payload['firstname'])
+      expect(user['lastname']).toEqual(payload['lastname'])
+    });
+
+    it('cannot create a user', async () => {
+      const payload = await factory.attrs('User', { firstname: undefined });
+
+      try {
+        await expect(
+          await (async () => userQueries.create({ payload }))(),
```

Comment:
This should do it:
`await expect(userQueries.create({ payload })).rejects.toThrow(new Error(`firstname` is required'))`
Instead of the try/catch

Reply:
Thank you @noisyscanner. This obviously will break the test because the error that is thrown is [`mongoose.Error.ValidatorError`](https://mongoosejs.com/docs/api.html#error-validation-js).

Reply:
Moreover I specifically added `try/catch` rather than `await expect` because I want to poke into the error that was thrown and run some assertions on it.

Reply:
> Thank you @noisyscanner. This obviously will break the test because the error that is thrown is [`mongoose.Error.ValidatorError`](https://mongoosejs.com/docs/api.html#error-validation-js).

And jest is not smart enough to know something like [**Liskov Substitution Principle**](https://stackify.com/solid-design-liskov-substitution-principle/) 😆